### PR TITLE
Declare return value in `Injector::make()` and improve psalm annotations

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  |❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: src/Injector.php [hunk 1]
```diff
@@ -91,20 +91,22 @@ public function invoke(callable $callable, array $arguments = [])
...
      * @throws ReflectionException
      *
-     * @return mixed object of the given class.
+     * @return object The object of the given class.
      *
      * @psalm-suppress MixedMethodCall
+     *
+     * @psalm-template T
+     * @psalm-param class-string<T> $class
+     * @psalm-return T
```
Comment (roxblnfk) [inline src/Injector.php:107]: I think you should use `@template` here (without `psalm-` prefix)

Comment (vjik) [reply src/Injector.php:107]: phpDocumentor don't support `@template`: https://docs.phpdoc.org/guide/references/phpdoc/tags/

Comment (roxblnfk) [reply src/Injector.php:107]: > phpDocumentor don't support `@template`: https://docs.phpdoc.org/guide/references/phpdoc/tags/

`@psalm-template` too.
What about phan and phpstan? The world is not limited to a psalm

Comment (vjik) [reply src/Injector.php:107]: > phan and phpstan

`@psalm-template` special for pslam. This annotations writed only for psalm.

I don't know how write annotations for phan and phpstan with support `class-string<T>`.

Comment (roxblnfk) [reply src/Injector.php:107]: > I don't know how write annotations for phan and phpstan with support class-string<T>.

Just use `@template T` and  `class-string<T>` https://github.com/phan/phan/blob/f36b6b9a2f4143a25f35ce94d712ceb0527e9d90/tests/files/src/0606_new_template.php

Don't see on PhpDocumentor. Static analizis is not for PhpDocumentor, it is for users

Comment (vjik) [reply src/Injector.php:107]: Will there be any problems with generating documentation if replce annotations

```
     * @param string $class name of the class to be created.
     * @return object The object of the given class.
     *
     * @psalm-template T
     * @psalm-param class-string<T> $class
     * @psalm-return T
```

to

```
     * @template T
     * @param class-string<T> $class name of the class to be created.
     * @return T The object of the given class.
```

?

Comment (roxblnfk) [reply src/Injector.php:107]: ¯\\\_(ツ)\_/¯

Comment (vjik) [reply src/Injector.php:107]: phpDocumentor generate incorrect documentation:

image

I think we should leave it as is:

```
     * @psalm-template T
     * @psalm-param class-string<T> $class
     * @psalm-return T
```

**Comment by samdark** [comment]
👍

**Pull request merged**